sim/team-validator.ts

```diff
@@ -2571,6 +2580,12 @@ export class TeamValidator {
 							moveSources.pomegEggMoves = [move.id];
 						} else if (learnedGen < 6) {
 							limitedEggMove = move.id;
+						} else {
+							let motherLearnset;
+							if (species.mother) motherLearnset = this.dex.species.getLearnset(toID(species.mother));
+							if (motherLearnset && !motherLearnset[move.id]) {
```

Comment:
```suggestion
						} else if (species.mother) {
							let motherLearnset = this.dex.species.getLearnset(toID(species.mother));
							if (!motherLearnset[move.id]) {
```
In that case, perhaps this would be clearer.

Reply:
I implemented the change to motherLearnset so that it doesn't always start out as undefined, but in the last line, removing the first condition makes the code prone to crashing, so I left that line as is.

Reply:
Well, I guess that works. (In my version I had moved, not removed, the `if (species.mother)` check.)